google/cloud/bigquery/model.py

```diff
@@ -184,6 +184,20 @@ def feature_columns(self) -> Sequence[standard_sql.StandardSqlField]:
             standard_sql.StandardSqlField.from_api_repr(column) for column in resource
         ]
 
+    @property
+    def transform_columns(self) -> Sequence[Dict[str, Any]]:
```

Comment:
Please make this return a sequence of your new TransformColumn objects instead of a sequence of dictionaries.